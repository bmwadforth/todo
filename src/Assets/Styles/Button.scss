@import "Variables";

button {
    width: 125px;
    font-size: 14px;
    border: 2px solid rgba($primary, 0.2);
    border-radius: $border-radius;
    background-color: $primary;
    color: #FFF;
    padding: 0.5em;

    transition: border $transition-time ease-in-out,
    background-color $transition-time ease-in-out,
    color $transition-time ease-in-out;

    &:hover {
        background-color: rgba(#FFF, 1);
        border: 2px solid rgba($primary, 0.8);
        color: #000;
        cursor: pointer;
    }

    &.button-success {
        background-color: $success;
        border: 2px solid rgba($success, 0.2);
        &:hover {
            background-color: rgba($success, 0);
            border: 2px solid rgba($success, 0.8);
            color: #000;
            cursor: pointer;
        }
    }

    &.button-warning {
        background-color: $warning;
        border: 2px solid rgba($warning, 0.2);
        &:hover {
            background-color: rgba($warning, 0);
            border: 2px solid rgba($warning, 0.8);
            color: #000;
            cursor: pointer;
        }
    }

    &.button-danger {
        background-color: $danger;
        border: 2px solid rgba($danger, 0.2);
        &:hover {
            background-color: rgba($danger, 0);
            border: 2px solid rgba($danger, 0.8);
            color: #000;
            cursor: pointer;
        }
    }
}
