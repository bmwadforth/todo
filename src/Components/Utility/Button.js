import React from "react";
import PropTypes from "prop-types";

export default function Button(props) {
    return (
        <button className={`button button-${props.status}`} onClick={props.onClick}
                type={props.type ? props.type : "button"}>{props.title}</button>
    )
}

Button.propTypes = {
    title: PropTypes.string.isRequired,
    status: PropTypes.oneOf(["primary", "success", "warning", "danger"]),
    type: PropTypes.string,
    onClick: PropTypes.func
};