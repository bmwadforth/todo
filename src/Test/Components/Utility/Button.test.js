import React from 'react';
import {shallow, configure} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Button from "../../../Components/Utility/Button";
configure({adapter: new Adapter()});

describe("Button Component", () => {
    it("Should Match Snapshot", () => {
        const wrapper = shallow(<Button title="Test" status="success"/>);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
