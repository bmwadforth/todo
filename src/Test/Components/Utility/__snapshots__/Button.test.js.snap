// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should Match Snapshot 1`] = `"<button class=\\"button button-success\\" type=\\"button\\">Test</button>"`;
